execAbort test: script is busy even after SCRIPT KILL command

PipeliningTest.execAbort() test sometimes fails in CI.

- https://app.circleci.com/pipelines/github/redis/jedis/1068/workflows/2c6b8e03-e85e-4c97-9759-233b5234d8fd/jobs/1272 
- https://app.circleci.com/pipelines/github/redis/jedis/1069/workflows/89107fbd-1e6d-46a9-baeb-1732e0baad2f/jobs/1274

In log:
```
[INFO] Running redis.clients.jedis.tests.PipeliningTest
[ERROR] Tests run: 37, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.734 s <<< FAILURE! - in redis.clients.jedis.tests.PipeliningTest
[ERROR] execAbort(redis.clients.jedis.tests.PipeliningTest)  Time elapsed: 0.04 s  <<< ERROR!
redis.clients.jedis.exceptions.JedisBusyException: BUSY Redis is busy running a script. You can only call SCRIPT KILL or SHUTDOWN NOSAVE.
	at redis.clients.jedis.tests.PipeliningTest.execAbort(PipeliningTest.java:739)
```

In summary:
```
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   PipeliningTest.execAbort:739 » JedisBusy BUSY Redis is busy running a script. ...
[INFO] 
[ERROR] Tests run: 883, Failures: 0, Errors: 1, Skipped: 0
```

Snapshot of code:

https://github.com/redis/jedis/blob/c9a04dc2a314c9a15b2cc25f2fdfce3df6e287aa/src/test/java/redis/clients/jedis/tests/PipeliningTest.java#L736-L740

Line 739 is getting executed only after `scriptKill()`.

--- a/src/test/java/redis/clients/jedis/tests/PipeliningTest.java
+++ b/src/test/java/redis/clients/jedis/tests/PipeliningTest.java
@@ -30,7 +30,7 @@ import redis.clients.jedis.Pipeline;
 import redis.clients.jedis.Response;
 import redis.clients.jedis.Tuple;
 import redis.clients.jedis.exceptions.AbortedTransactionException;
-import redis.clients.jedis.exceptions.JedisConnectionException;
+import redis.clients.jedis.exceptions.JedisBusyException;
 import redis.clients.jedis.exceptions.JedisDataException;
 import redis.clients.jedis.tests.commands.JedisCommandTestBase;
 import redis.clients.jedis.util.SafeEncoder;
@@ -735,8 +735,15 @@ public class PipeliningTest extends JedisCommandTestBase {
       assertSame(AbortedTransactionException.class, ex.getClass());
     } finally {
       try {
-        jedis.scriptKill();
-        scriptKillWait();
+        String status = jedis.scriptKill();
+        // https://github.com/redis/jedis/issues/2656
+        if ("OK".equalsIgnoreCase(status)) {
+          scriptKillWait();
+        } else {
+          // #2656: Checking if this status is actually 'OK' when error occurs in next command.
+          org.apache.logging.log4j.LogManager.getLogger().error(
+              String.format("Status if SCRIPT KILL command is \"%s\"", status));
+        }
       } finally {
         jedis.configSet(luaTimeLimitKey, luaTimeLimit);
       }
@@ -751,15 +758,14 @@ public class PipeliningTest extends JedisCommandTestBase {
           try {
             pingJedis.ping();
             return; // wait is over
-          } catch (JedisDataException jde) {
+          } catch (JedisBusyException busy) {
             --attemptLeft;
-            Thread.sleep(10);
-          } catch (JedisConnectionException jce) {
-            break; // try new connection
+            Thread.sleep(10); // BUSY, waiting for some time
           }
         }
       } catch (Exception any) {
-        // catching everything to try again
+        --attemptLeft;
+        // try new connection
       }
     }
   }

--- a/src/test/java/redis/clients/jedis/tests/PipeliningTest.java
+++ b/src/test/java/redis/clients/jedis/tests/PipeliningTest.java
@@ -30,6 +30,7 @@ import redis.clients.jedis.Pipeline;
 import redis.clients.jedis.Response;
 import redis.clients.jedis.Tuple;
 import redis.clients.jedis.exceptions.AbortedTransactionException;
+import redis.clients.jedis.exceptions.JedisConnectionException;
 import redis.clients.jedis.exceptions.JedisDataException;
 import redis.clients.jedis.tests.commands.JedisCommandTestBase;
 import redis.clients.jedis.util.SafeEncoder;
@@ -735,8 +736,30 @@ public class PipeliningTest extends JedisCommandTestBase {
     } finally {
       try {
         jedis.scriptKill();
+        scriptKillWait();
       } finally {
         jedis.configSet(luaTimeLimitKey, luaTimeLimit);
+      }
+    }
+  }
+
+  private void scriptKillWait() {
+    int attemptLeft = 10;
+    while (attemptLeft > 0) {
+      try (Jedis pingJedis = createJedis()) {
+        while (attemptLeft > 0) {
+          try {
+            pingJedis.ping();
+            return; // wait is over
+          } catch (JedisDataException jde) {
+            --attemptLeft;
+            Thread.sleep(10);
+          } catch (JedisConnectionException jce) {
+            break; // try new connection
+          }
+        }
+      } catch (Exception any) {
+        // catching everything to try again
       }
     }
   }

